app/assets/javascripts/pretty-text/addon/engines/discourse-markdown-it.js

```diff
@@ -423,6 +423,7 @@ function unhoistForCooked(hoisted, cooked) {
         found = true;
         return hoisted[key];
       });
+      delete hoisted[key];
```

Comment:
Not strictly related to this, but to the `key` in `new RegExp(key, "g")`: should not this be escaped?

Reply:
Why? The key is a GUID generated here https://github.com/discourse/discourse/blob/4843c4a44ff420269bed56700d5bc353e6181255/app/assets/javascripts/pretty-text/addon/engines/discourse-markdown-it.js#L136-L137